### DISCUSSION TITLE
why imgui showing linking error

### DISCUSSION THREAD

**Discussion by neko-san12XX** [ocornut/imgui#6449] [New Users Build/Link/Run issues ONLY!]
(Click "Preview" above ^ to turn URL into clickable links)

1. FOR FIRST-TIME USERS ISSUES COMPILING/LINKING/RUNNING or LOADING FONTS,  please use [GitHub Discussions](https://github.com/ocornut/imgui/discussions).

2. PLEASE CAREFULLY READ: [FAQ](https://github.com/ocornut/imgui/blob/master/docs/FAQ.md)

3. PLEASE CAREFULLY READ: [Contributing Guidelines](https://github.com/ocornut/imgui/blob/master/docs/CONTRIBUTING.md)

4. PLEASE MAKE SURE that you have: read the FAQ; explored the contents of `ShowDemoWindow()` including the Examples menu; searched among Issues; used your IDE to search for keywords in all sources and text files; and read the links above.

5. Be mindful that messages are being sent to the e-mail box of "Watching" users. Try to proof-read your messages before sending them. Edits are not seen by those users.

6. Delete points 1-6 and PLEASE FILL THE TEMPLATE BELOW before submitting your issue.

Thank you!

----

_(you may also go to Demo>About Window, and click "Config/Build Information" to obtain a bunch of detailed information that you can paste here)_

**Version/Branch of Dear ImGui:**

Version: XXX
Branch: XXX _(master/viewport/docking/etc.)_

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_XXX.cpp + imgui_impl_XXX.cpp _(or specify if using a custom engine/back-end)_
Compiler: XXX _(if the question is related to building or platform specific features)_
Operating System: XXX

**My Issue/Question:**

XXX _(please provide as much context as possible)_

**Screenshots/Video**

XXX _(you can drag files here)_

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
// Here's some code anyone can copy and paste to reproduce your issue
ImGui::Begin("Example Bug");
MoreCodeToExplainMyIssue();
ImGui::End();
```

**Comment by neko-san12XX** [comment]
i included imgui for opengl. everything looks fine but for some reason, it is showing linking error. why?

**Reply by ocornut** [reply]
Hello,
You aren't going to get help if you don't provide more details..

**Comment by pvince999** [comment]
Problem linking imgui (with GLAD ?).
_________________
[100%] Linking CXX executable Imgui1/Imgui1
Undefined symbols for architecture x86_64:
  "ImFontAtlas::AddFontFromFileTTF(char const*, float, ImFontConfig const*, unsigned short const*)", referenced from:
      _main in main.cpp.o
  "ImFontAtlas::GetTexDataAsRGBA32(unsigned char**, int*, int*, int*)", referenced from:
      _main in main.cpp.o
  "ImFontConfig::ImFontConfig()", referenced from:
      _main in main.cpp.o
  "ImGui::GetDrawData()", referenced from:
      _main in main.cpp.o
  "ImGui::CreateContext(ImFontAtlas*)", referenced from:
      _main in main.cpp.o
  "ImGui::DestroyContext(ImGuiContext*)", referenced from:
      _main in main.cpp.o
  "ImGui::ShowDemoWindow(bool*)", referenced from:
      _main in main.cpp.o
  "ImGui::GetIO()", referenced from:
      _main in main.cpp.o
      mouse_callback(GLFWwindow*, double, double) in main.cpp.o
      mouseButtonCallback(GLFWwindow*, int, int, int) in main.cpp.o
  "ImGui::Render()", referenced from:
      _main in main.cpp.o
  "ImGui::NewFrame()", referenced from:
      _main in main.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1
-----------------------
If you need more info, please shout.

**Reply by ocornut** [reply]
You are not linking with compiled imgui files. Add them to your project.

**Reply by pvince999** [reply]
Thank  you!
Ignorance on my part.

**Reply by ocornut** [reply]
It’s among the most common issue learning C++ users are having with librairies, mistaking includes with object files. Once you understand your build process it’ll become mostly a non-issue :)
